### DISCUSSION TITLE
Lingering Rg calls from VIM

### DISCUSSION THREAD

**Discussion by hqkhan** [BurntSushi/ripgrep#1641] [General] [answered]
I'm using `wsltty` as my terminal in Windows 10. In it, I'm using `neovim (nvim)` as my editor. I recently discovered `ripgrep` and wanted to use it in `nvim`. It works great in terms of performance but I'm seeing lingering `rg` calls in the task manager. Apparently, the `rg` calls are running into an error and end up not quitting. Is there a solution to this? 

After some digging around, people mentioned that switching from WSL 1 to WSL 2 fixed it for them. I am not able to make this switch. Is there a fix for WSL 1?

This is what it looks like. These errors are all over the place:
```
hkhan@MM243812-PC:~/.vim$ rg onedark
thread '<unnamed>' panicked at 'assertion failed: `(left == right)`
  left: `22`,
 right: `4`', src/libstd/sys/unix/thread.rs:172:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
plugged/onedark.vim/CONTRIBUTING.md
1:# Contributing to onedark.vim
11:onedark.vim's shared color definitions file `autoload/onedark.vim` and its companion Xresources color palette file are built using a rudimentary templating and build sys
tem that allows color definitions to live in a single, central file. (Additional terminal color palette files are themselves generated from the template-produced Xresources
 file.)
19:| `autoload/onedark.vim`                       | `build/templates/autoload.template.vim`  |
55:For the "Language-Specific Highlighting" portion of onedark.vim, blocks for each language should be organized alphabetically ("Markdown" comes before "PHP").
61:Thanks very much for contributing to onedark.vim!
```

**Comment by BurntSushi** [comment, marked as answer]
You can try `-j1` to force single threaded mode. Otherwise, AFAIK, WSL1 is broken.

**Reply by hqkhan** [reply]
Would the command be `rg -j1 onedark` instead? It doesn't work for me. Tried `rg -j1` to maybe turn it into single threaded mode but that didn't work either. By "didn't work", I mean that there is no output:

```
hkhan@MM243812-PC:~$ rg -j1 onedark
hkhan@MM243812-PC:~$ rg -j1 onedark .
```

Ah, I see. What is an alternative to this? Using a different terminal emulator?

**Reply by BurntSushi** [reply]
I don't know, sorry. I'm not a Windows user. From what I've heard, WSL1 is badly borked. Maybe try cygwin? Or just standard powershell? I've heard from others that `-j1` has resolved it I think (search the issue tracker). Idk why it isn't working for you. I'd recommend debugging more. Try searching a single file for example.

**Reply by hqkhan** [reply]
Ok got it. I'll try to dig more into it. Thanks!
